dali/operators/image/paste/multipaste.cc

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/paste/multipaste.h"
+#include "dali/kernels/imgproc/paste/paste.h"
+#include "dali/core/tensor_view.h"
+
+namespace dali {
+namespace {
+
+template <typename Out, typename In>
+using TheKernel = kernels::PasteCpu<Out, In>;
+
+}  // namespace
+
+
+DALI_SCHEMA(MultiPaste)
+.DocStr(R"code(Performs multiple (K * batch_size) pastes from image batch to an output
+
+This operator can also change the type of data.)code")
+
+.NumInput(5, 6)
+.InputDox(0, "images", "3D TensorList", R"code(Batch of input images.
+
+Assumes HWC layout.)code")
+.InputDox(1, "in_ids", "1D TesorList of shape [K] and type int",
+R"code(Indexes from what inputs to paste data in each iteration.)code")
+.InputDox(2, "in_anchors", "2D TesorList of shape [K, 2] and type int",
+R"code(Absolute values of LU corner of the selection for each iteration.)code")
+.InputDox(3, "in_shapes", "2D TesorList of shape [K, 2] and type int",
+R"code(Absolute values of size of the selection for each iteration.)code")
+.InputDox(4, "out_anchors", "2D TesorList of shape [K, 2] and type int",
+R"code(Absolute values of LU corner of the paste for each iteration.)code")
+.InputDox(5, "out_ids", "1D TesorList of shape [K] and type int",
+R"code(Indexes to what outputs to paste data in each iteration.
```

Comment:
This kind of mapping is both redundant and dangerous.
Redundancy stems from the fact that you can just permute the input to get the desired result.
Danger comes in two flavors:
- race condition: two samples write to the same index
- missing sample: some output is never specified
It also complicates batch size reduction - and when we finally allow the batch size to change across operators in the pipeline, just specifying as many _input_ parameter sets as there are desired _output_ samples is the way to go.

Reply:
It is very much possible to have a specific use case, where one would like to perform no paste to one of the outputs - it should be left blank in such case, as no pastes target it. As for race condition, there are 3 worker strategies:
- if no_intersections is assumed, a separate work item is launched for each paste
- if no_intersections is not assumed, but out_ids is not given, one work item runs all the pastes for a given output
- if no_intersections is not assumed and out_ids is given, there is no easy way to parallelize, so one worker runs all the pastes

Reply:
[EDIT After reading last comments] Ok now I understand. So the shapes of the tensors in TensorList need not be uniform, so I can put any number (for example 0) pastes for each index separately, and that's how many pastes will run to the corresponding output.

Got rid of this argument and fixed tests accordingly